ultraviolet/shared/src/test/scala/ultraviolet/SyntaxTests.scala

```diff
@@ -0,0 +1,57 @@
+package ultraviolet
+
+import syntax.*
+import scala.util.Random
+
+class SyntaxTests extends munit.FunSuite {
+
+  // TODO: Should the input be prefixed by '#'?
+  // TODO: Do we want shorthand notation (i.e. #000 to mean #000000)?
+  test("hex interpolator") {
+    assertEquals(hex"00FF00", vec3(0f, 1f, 0f))
+    assertEquals(hex"ff00ff", vec3(1f, 0f, 1f))
+    val (hex1, hex2, hex3) = ("00", "ff", "00")
+    assertEquals(hex"$hex1$hex2$hex3", vec3(0f, 1f, 0f))
+
+    intercept[IllegalArgumentException](hex"00000"): Unit
+    intercept[IllegalArgumentException](hex"0000000"): Unit
+    intercept[IllegalArgumentException](hex"gggggg"): Unit
+  }
+
+  test("hexa interpolator") {
+    assertEquals(hexa"00FF00FF", vec4(0f, 1f, 0f, 1f))
+    assertEquals(hexa"ff00ff00", vec4(1f, 0f, 1f, 0f))
+    val (hex1, hex2, hex3, hex4) = ("00", "ff", "00", "ff")
+    assertEquals(hexa"$hex1$hex2$hex3$hex4", vec4(0f, 1f, 0f, 1f))
+
+    intercept[IllegalArgumentException](hexa"0000000"): Unit
+    intercept[IllegalArgumentException](hexa"000000000"): Unit
+    intercept[IllegalArgumentException](hexa"gggggggg"): Unit
+  }
+
+  test("rgb interpolator") {
+    assertEquals(rgb"0,0,0", vec3(0f, 0f, 0f))
+    assertEquals(rgb"255,0,255", vec3(1f, 0f, 1f))
+    val (int1, int2, int3) = (0, 255, 0)
+    assertEquals(rgb"$int1,$int2,$int3", vec3(0f, 1f, 0f))
+
+    intercept[IllegalArgumentException](rgb"0,0"): Unit
+    intercept[IllegalArgumentException](rgb"0,0,0,0"): Unit
+    intercept[IllegalArgumentException](rgb"0, 0, 0"): Unit
```

Comment:
The goal here should be to be as friendly / helpful as possible, while making a pragmatic call on the amount of effort it takes to do the work, i.e. is the cost vs value acceptable.

In this case it could be:

1. It's easy to do and reasonably helpful, so let's do it.
2. It's a pain to do, and actually helpful, but not so helpful that it's worth the hassle.
3. ...but maybe the cost of adding a descriptive error is acceptable instead? "rgb values cannot contain whitespace, and should take the form `rgb"1,64,255"`?